### PR TITLE
SONAR-6353 new metric days_since_last_commit

### PR DESCRIPTION
**Pull request by simonbrandhof**
https://jira.sonarsource.com/browse/SONAR-6353

### CODE REVIEW

**Comment by julienlancelot** [comment]
LGTM

**Comment by sns-seb** [comment]
@simonbrandhof LGTM

File: server/sonar-server/src/main/java/org/sonar/server/computation/source/LastCommitVisitor.java [hunk 1]
```diff
@@ -35,12 +36,16 @@
...
   private final BatchReportReader reportReader;
   private final MeasureRepository measureRepository;
   private final Metric lastCommitDateMetric;
+  private final Metric daysSinceLastCommitDateMetric;
+  private final System2 system2;
 
   public LastCommitVisitor(BatchReportReader reportReader, MetricRepository metricRepository,
-    MeasureRepository measureRepository) {
+    MeasureRepository measureRepository, System2 system2) {
     super(CrawlerDepthLimit.LEAVES, POST_ORDER, new SimpleStackElementFactory<LastCommit>() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/source/LastCommitVisitor.java]: MAJOR Make this a named "static" inner class. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

File: server/sonar-server/src/main/java/org/sonar/server/computation/source/LastCommitVisitor.java [hunk 2]
```diff
@@ -35,12 +36,16 @@
 
 public class LastCommitVisitor extends PathAwareVisitorAdapter<LastCommitVisitor.LastCommit> {
 
+  private static final long MILLISECONDS_PER_DAY = 1000 * 60 * 60 * 24;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/source/LastCommitVisitor.java]: MAJOR Cast one of the operands of this multiplication operation to a "long". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 1 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* CRITICAL Class org.sonar.api.database.model.Snapshot removed (org.codehaus.sonar:sonar-plugin-api) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)

**Pull request closed without merging**